cmd/mesh-bootstrap/main.go

```diff
@@ -115,14 +122,55 @@ func main() {
 	// We need to declare the newDHT function ahead of time so we can use it in
 	// the libp2p.Routing option.
 	var kadDHT *dht.IpfsDHT
-	newDHT := func(h host.Host) (routing.PeerRouting, error) {
-		var err error
-		dhtDir := getDHTDir(config)
-		kadDHT, err = p2p.NewDHT(ctx, dhtDir, h)
-		if err != nil {
-			log.WithField("error", err).Fatal("could not create DHT")
+	var newDHT func(h host.Host) (routing.PeerRouting, error)
+
+	// TODO(oskar) - Figure out why returning an anonymous function from
```

Comment:
@opaolini can you elaborate on what you are talking about here?

Reply:
So previously I had here:
```go
var kadDHT *dht.IpfsDHT
newDHT := getNewDHT(ctx, config, kadDHT)
```

However this caused panics with nil pointer dereference for `*dht.IpfsDHT` Method, I did not dive too deep into what exactly happens there.

Reply:
It's hard for me to understand exactly what's happening without looking at the state of the codebase when the error occurred. I'm okay with just leaving it for now.